zerver/views/users.py

```diff
@@ -489,3 +489,23 @@ def get_profile_backend(request: HttpRequest, user_profile: UserProfile) -> Http
         result['max_message_id'] = messages[0].id
 
     return json_success(result)
+
+@has_request_variables
+def get_subscription_backend(request: HttpRequest, user_profile: UserProfile,
+                             user_id: int=REQ(validator=check_int),
+                             stream_id: int=REQ(validator=check_int)
+                             ) -> HttpResponse:
+    target_user = access_user_by_id(user_profile, user_id, read_only=True)
+    subscriptions, _ = gather_subscriptions(target_user)
```

Comment:
This seems really expensive; the whole goal here is to just check if the user is subscribed.

We should validate access to the stream via `access_stream_by_id`, and then if that succeeds, use `subscribed_to_stream` from `zerver/lib/streams.py` to do the check.